api/v1alpha1/loadbalancer_types.go

```diff
@@ -53,17 +53,32 @@ const (
 
 // ConsistentHash defines the configuration related to the consistent hash
 // load balancer policy
+// +kubebuilder:validation:XValidation:rule="self.type == 'Header' ? has(self.header) : !has(self.header)",message="If consistent hash type is header, the header field must to be set."
 type ConsistentHash struct {
-	Type ConsistentHashType `json:"type"`
+	Type   ConsistentHashType `json:"type"`
+	Header Header             `json:"header,omitempty"`
```

Comment:
Needs to be a ptr type since it's an optional field

Reply:
This should be fixed on the most recent push